### PR TITLE
Follow-up actions for the GeoServer interceptor

### PR DESCRIPTION
**Pull request by dnlkoch**
Within this PR you'll find some enhancements to the GeoServer interceptor mentioned in #144:

* Make use of enumerations where possible (e.g. for the OGC nomenclature).
* Refactored `getMostSpecificRule()` method (minimized database requests).
* Enabled more logging for easier debugging.
* Add some tests for the `GeoServerInterceptorService`.

### CODE REVIEW

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java [hunk 1]
```diff
@@ -144,124 +151,169 @@ public Response interceptGeoServerRequest(HttpServletRequest request)
...
-			return result.get(0);
+		LOG.trace("Evaluating the given rules for the most specific one:");
+
+		// create the predicate for finding the most specific rule out of
+		// the given rules (conditions in descending specific order).
+		// note: we don't need to check for request event (request/response)
+		// and the service type at all as the rule candidates are filtered by
+		// the DAO method already. the last check for service specificity found
+		// here is a fallback only.
+		Predicate<InterceptorRule> condition = new Predicate<InterceptorRule>() {
```
Comment (marcjansen) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java]: Can we put this into aclass-property or a constant?

**Comment by marcjansen** [comment]
I added one note, otherwise this looks great to me.

Are all the notes from #144 resolved now? 

Thanks & please merge (with or without taking my suggestion).

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java (hunk 1 above)
Comment (marcjansen) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java]: Ignore the above :smiling_imp:

**Comment by dnlkoch** [comment]
Thanks @marcjansen for your review!

Not all notes are resolved inside this PR, we are still missing:
* A global proxy ([see here](https://github.com/terrestris/shogun2/pull/144#discussion-diff-55359142))
* Additionally abstract classes that provide the default implementations for the service interceptors ([see here](https://github.com/terrestris/shogun2/pull/144#issuecomment-193805000))

But I would favour to address this issues (especially the first one) in individual PRs.

**Comment by marcjansen** [comment]
Go ahead, please merge!

**Pull request merged**